main.go

```diff
@@ -22,17 +23,26 @@ import (
 )
 
 var (
-	pass = color.FgGreen
-	skip = color.FgYellow
-	fail = color.FgHiRed
+	pass   = color.FgGreen
+	skip   = color.FgYellow
+	fail   = color.FgHiRed
+	ignore = flag.Bool("skipnotest", false, "skip packages with no test files")
 )
 
 const paletteEnv = "GOTEST_PALETTE"
 
 func main() {
+	flag.Parse()
 	setPalette()
 	enableOnCI()
-	os.Exit(gotest(os.Args[1:]))
+
+	var startingIndex = 1
+	if strings.Contains(os.Args[1], "-skipnotest") {
+		startingIndex++
+	}
+
+	os.Exit(gotest(os.Args[startingIndex:]))
```

Comment:
You can use flag.Args instead, see https://golang.org/pkg/flag/#Args

It returns the non-flag arguments.

Reply:
Wonderful!

Reply:
I think using `flag.Args` will prevent flags that we intend to pass to `go test` from being forwarded.

Reply:
@subtlepseudonym 
No, It doesn't I tested it.
for example:
`gotest -skipnotest ./... -cover
`